include/dlaf/eigensolver/tridiag_solver/merge.h

```diff
@@ -0,0 +1,1018 @@
+  TileCollector tc{i_begin, i_end};
+
+  auto sender =
+      ex::when_all(ex::keep_future(k_fut), ex::keep_future(rho_fut), ex::when_all_vector(tc.read(d)),
+                   ex::when_all_vector(tc.read(z)), ex::when_all_vector(tc.readwrite(evals)),
+                   ex::when_all_vector(tc.readwrite(evecs)));
+
+  ex::ensure_started(
+      di::transform<false>(di::Policy<Backend::MC>(), std::move(rank1_fn), std::move(sender)));
+}
+
+// References:
+// - lapack 3.10.0, dlaed3.f, line 293
+// - LAPACK Working Notes: lawn132, Parallelizing the Divide and Conquer Algorithm for the Symmetric
+//   Tridiagonal Eigenvalue Problem on Distributed Memory Architectures, 4.2 Orthogonality
+template <class T>
+void formEvecs(SizeType i_begin, SizeType i_end, pika::shared_future<SizeType> k_fut,
+               Matrix<const T, Device::CPU>& diag, Matrix<const T, Device::CPU>& z,
+               Matrix<T, Device::CPU>& ws, Matrix<T, Device::CPU>& evecs) {
+  namespace ex = pika::execution::experimental;
+  namespace di = dlaf::internal;
+
+  using ConstTileType = matrix::Tile<const T, Device::CPU>;
+  using TileType = matrix::Tile<T, Device::CPU>;
+
+  const matrix::Distribution& distr = evecs.distribution();
+
+  for (SizeType i_tile = i_begin; i_tile <= i_end; ++i_tile) {
+    SizeType i_subm_el = distr.globalTileElementDistance<Coord::Row>(i_begin, i_tile);
+    for (SizeType j_tile = i_begin; j_tile <= i_end; ++j_tile) {
+      SizeType j_subm_el = distr.globalTileElementDistance<Coord::Col>(i_begin, j_tile);
+      auto mul_rows = [i_subm_el, j_subm_el](SizeType k, const ConstTileType& diag_rows,
+                                             const ConstTileType& diag_cols,
+                                             const ConstTileType& evecs_tile, const TileType& ws_tile) {
+        if (i_subm_el >= k || j_subm_el >= k)
+          return;
+
+        SizeType nrows = std::min(k - i_subm_el, evecs_tile.size().rows());
+        SizeType ncols = std::min(k - j_subm_el, evecs_tile.size().cols());
+
+        for (SizeType j = 0; j < ncols; ++j) {
+          for (SizeType i = 0; i < nrows; ++i) {
+            T di = diag_rows(TileElementIndex(i, 0));
+            T dj = diag_cols(TileElementIndex(j, 0));
+            T evec_el = evecs_tile(TileElementIndex(i, j));
+            T& ws_el = ws_tile(TileElementIndex(i, 0));
+
+            // Exact comparison is OK because di and dj come from the same vector
+            T weight = (di == dj) ? evec_el : evec_el / (di - dj);
+            ws_el = (j == 0) ? weight : weight * ws_el;
+          }
+        }
+      };
+      ex::when_all(ex::keep_future(k_fut), diag.read_sender(GlobalTileIndex(i_tile, 0)),
+                   diag.read_sender(GlobalTileIndex(j_tile, 0)),
+                   evecs.read_sender(GlobalTileIndex(i_tile, j_tile)),
+                   ws.readwrite_sender(GlobalTileIndex(i_tile, j_tile))) |
+          di::transform(di::Policy<Backend::MC>(), std::move(mul_rows)) | ex::ensure_started();
```

Comment:
```suggestion
      ex::when_all(ex::keep_future(k_fut), diag.read_sender(GlobalTileIndex(i_tile, 0)),
                   diag.read_sender(GlobalTileIndex(j_tile, 0)),
                   evecs.read_sender(GlobalTileIndex(i_tile, j_tile)),
                   ws.readwrite_sender(GlobalTileIndex(i_tile, j_tile))) |
          di::transformDetached(di::Policy<Backend::MC>(), std::move(mul_rows));
```

and same elsewhere. Alternatively:

```suggestion
      ex::start_detached(ex::when_all(ex::keep_future(k_fut), diag.read_sender(GlobalTileIndex(i_tile, 0)),
                   diag.read_sender(GlobalTileIndex(j_tile, 0)),
                   evecs.read_sender(GlobalTileIndex(i_tile, j_tile)),
                   ws.readwrite_sender(GlobalTileIndex(i_tile, j_tile))) |
          di::transform(di::Policy<Backend::MC>(), std::move(mul_rows)));
```

These are functionally equivalent to what you have, but semantically clearer since you're not using the sender returned from `ensure_started`.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/436/commits/966bd5e36d84d2b231bbaab91fcaaae43140d0a1